chalicelib/action.py

```diff
@@ -465,10 +441,76 @@ def perform_action(self):
             )
             return ""
 
-        self._create_list_message(data=list_data)
+        self._create_list_message(data=list_data, period_data=period_data)
         self.slack.post_message(message="From timereport", channel=self.user_id)
         return ""
 
+    def _get_events(self, date_str):
+        return get_list_data(
+            f"{self.config['backend_url']}", self.user_id, date_str=date_str
+        )
+
+    def _create_list_message(self, data, period_data) -> None:
+        """
+        Create the slack block message layout for list action
+        """
+        data = json.loads(data)
+
+        start_date = data[0].get("event_date")
+        end_date = data[-1].get("event_date")
+
+        self.slack.add_section_block(
+            text=f"Reported time for period *{start_date}:{end_date}*",
+        )
+        self._show_period_data(list_data=data, period_data=period_data)
+        self.slack.add_divider_block()
+
+        for event in data:
+            event_date = event.get("event_date")
+            reason = event.get("reason")
+            hours = event.get("hours")
+            self.slack.add_section_block(
+                text=f"Date: *{event_date}*\nReason: *{reason}*\nHours: *{hours}*"
+            )
+            self.slack.add_divider_block()
+
+    def _show_period_data(self, list_data, period_data) -> Dict[str, int]:
+        if not period_data:
+            self.slack.add_section_block(text="No information about worked hours")
+            return
+
+        total_per_type = defaultdict(int)
```

Comment:
Care to explain why we need to use `defaultdict`?

Reply:
With defaultdict we can do this:
```python
total_per_type[event.get("reason")] += hours
```

Without we have to do something like this:
```python
reason = event.get("reason")
total_per_type[reason] = total_per_type.get(reason, 0) + hours
```

I prefer the first one for readability of the code